sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp

```diff
@@ -12,6 +12,11 @@
 #include "azure/core/http/http.hpp"
 #include "azure/core/http/policies/policy.hpp"
 #include "azure/core/http/transport.hpp"
+#include "azure/core/platform.hpp"
+
+#if defined(AZ_PLATFORM_LINUX) && OPENSSL_VERSION_NUMBER >= 0x00905100L
+#define SUPPORT_SETTING_CAPATH
```

Comment:
1. `SUPPORT_SETTING_CAPATH` is too generic of a name - it could collide with customer code. Normally, it would be named something like `_azure_SUPPORT_SETTING_CAPATH`. But I think we should just inline the condition without defining anything.
2. `OPENSSL_VERSION_NUMBER` is defined in `openssl/opensslv.h`. Are we sure that even if `OPENSSL_VERSION_NUMBER >= 0x00905100L`, the `OPENSSL_VERSION_NUMBER` is always defined in the place where users include `curl_transport.hpp`? It does not look like this file does even include `curl.h` - it is only the `.cpp` file which includes it.
I think it should be safe to `#include <openssl/opensslv.h>` only inside an `#if defined(AZ_PLATFORM_LINUX)`.
@LarryOsterman, additional thoughts?
It is possible to build the SDK without any transport support, and then openssl dependency won't be installed, however I think it is ok to include it in this file, because `curl_transport` is built conditionally, and looking at `win_http_transport`, I think it is expectation for the customer to not include `win_http_transport.hpp`, if they did not build the SDK with WinHTTP support, otherwise even a single `#include <transport_that_was_not_built>` directive may fail to compile one way or another.

Reply:
My thinking for this was to put the openssl version check as simply `#if defined(OPENSSL_VERSION_NUMBER)` - if the macro is defined, it implies that openssl is configured. 

But it's complicated because CURL is configured with a specific TLS back end for each platform, and that's a part of how customers configure curl - the azure SDK doesn't have a lot of ability to control that.

Reply:
Customer may have the right openssl version installed, but when they include `curl_transport.hpp`, `OPENSSL_VERSION_NUMBER` is still not defined. They'd have to make sure they include `openssl/opensslv.h` BEFORE `curl_transport.hpp`, and that's fragile and complicated.

Reply:
Thanks for the discussion and comment. They are valuable as I don't have much experience in the development of library.
1. I prefer renaming `SUPPORT_SETTING_CAPATH` to inline the condition as it will much easier for further changes - just like the purpose of `AZ_PLATFORM_LINUX`
2. I have updated the condition of defining `SUPPORT_SETTING_CAPATH`. Now `#if defined(AZ_PLATFORM_LINUX)` will be checked first. As @antkmsft has specified that 
```
I think it should be safe to #include <openssl/opensslv.h> only inside an #if defined(AZ_PLATFORM_LINUX).
```
then `openssl/opensslv.h` is included to provide `OPENSSL_VERSION_NUMBER` for checking.

`curl.cpp` is a good reference for above changes.